Components/Settings.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NekoClient.Logging;
+using MU3.Util;
+
+namespace UnityParrot.Components
+{
+    public enum AnalogControlStyle
+    {
+        Mouse = 0,
+        Touchpad = 1,
+        Buttons = 2
+    }
+
+    [Serializable]
+    public class Settings
+    {
+        public string buttonSettings_begin;
```

Comment:
Please follow coding conventions.

I'd suggest something like
```csharp
class ButtonSettings
{
    public string Begin { get; set; }
}

public class AnalogSettings
{
    public string ControlStyle { get; set; }
}

public class Settings
{
     public ButtonSettings Buttons { get; set; }
     public AnalogSettings Analogs { get; set; }
}
```

Also, unity's KeyCode.Something keys can be serialized just fine, the [Serializable] header is not needed

Reply:
For some reason I'm having issues with ToJson properly serializing a nested class here. It also does not serialize properties. That being said I changed the above flat settings to have proper variable names.